ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx

```diff
@@ -386,60 +364,21 @@ function NetworkGraphPage() {
                     <Divider component="div" />
                 </>
             )}
-            <PageSection
-                className="network-graph"
-                variant={hasClusterNamespaceSelected ? 'default' : 'light'}
-                padding={{ default: 'noPadding' }}
-            >
-                {hasReadAccessForGenerator && !hasClusterNamespaceSelected && (
-                    <Drawer isExpanded={simulation.isOn} isInline>
-                        <DrawerContent
-                            panelContent={
-                                <DrawerPanelContent
-                                    className="cluster-simulation-drawer-panel"
-                                    isResizable
-                                    defaultSize="500px"
-                                >
-                                    <Button
-                                        className="pf-topology-side-bar__dismiss"
-                                        variant="plain"
-                                        onClick={closeSimulatorSidebar}
-                                        aria-label="Close sidebar"
-                                    >
-                                        <TimesIcon />
-                                    </Button>
-                                    <NetworkPolicySimulatorSidePanel
-                                        simulator={simulator}
-                                        setNetworkPolicyModification={setNetworkPolicyModification}
-                                        scopeHierarchy={scopeHierarchy}
-                                        scopeDeploymentCount={deploymentCount ?? 0}
-                                    />
-                                </DrawerPanelContent>
-                            }
-                        >
-                            <DrawerContentBody>
-                                <EmptyUnscopedState />
-                            </DrawerContentBody>
-                        </DrawerContent>
-                    </Drawer>
-                )}
-                {models.activeModel.nodes.length > 0 &&
-                    models.extraneousModel.nodes.length > 0 &&
-                    !isLoading &&
-                    hasClusterNamespaceSelected && (
-                        <NetworkGraphContainer
-                            models={models}
-                            edgeState={edgeState}
-                            displayOptions={displayOptions}
-                            simulation={simulation}
-                            clusterDeploymentCount={deploymentCount || 0}
-                            scopeHierarchy={scopeHierarchy}
-                        />
-                    )}
-                {isLoading && (
+            <PageSection className="network-graph" padding={{ default: 'noPadding' }}>
+                {isLoading ? (
```

Comment:
oh is there a reason why the other conditions are left out here? `models.activeModel.nodes.length` and `models.extraneousModel.nodes.length`. i think we added those in to catch edge cases (since they result from two different network calls)

Reply:
Good catch, I left those out since when that condition triggered the result would be a plain grey screen (there was no other component rendered for that condition). The new behavior is almost the same, except that the graph is rendered with the overlays:
![image](https://github.com/stackrox/stackrox/assets/1292638/6baa7d6e-1969-43d7-bdb0-01e042d65dd0)

I think an ideal follow up would be a separate "no data" view, different from the blank grey screen and different from the "Nothing to render yet" screen that tells the user to select a cluster + namespace.

Reply:
Oh actually I found another bug testing this right now that apparently happens on staging as well. If you select a namespace with deployments, deselect it, then select a new namespace with zero deployments, you'll get a view with stale deployment data instead of the empty view.

Reply:
https://issues.redhat.com/browse/ROX-21672